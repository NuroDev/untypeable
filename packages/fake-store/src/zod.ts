export * from "./cart/cart.validators";
export * from "./products/products.validators";
