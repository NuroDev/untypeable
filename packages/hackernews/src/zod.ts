export * from "./items/items.validators";
