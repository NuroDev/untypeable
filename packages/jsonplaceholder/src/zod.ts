export * from "./comment/comment.validators";
export * from "./post/post.validators";
