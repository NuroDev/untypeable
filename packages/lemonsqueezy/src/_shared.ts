export enum LemonsqueezyDataType {
  checkouts = "checkouts",
  discounts = "discounts",
  files = "files",
  license_key_instances = "license-key-instances",
  license_keys = "license-keys",
  order_items = "order-items",
  orders = "orders",
  products = "products",
  stores = "stores",
  subscriptions = "subscriptions",
  users = "users",
  variants = "variants",
}

export interface LemonSqueezyResponse<
  TData,
  TLinks = {
    self: string;
  }
> {
  data: TData;
  errors?: Array<{
    detail: string;
    status: string | number;
    title: string;
  }>;
  jsonapi: {
    version: string;
  };
  links: TLinks;
}
