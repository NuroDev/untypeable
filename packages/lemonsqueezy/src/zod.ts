export { DataType } from "./_shared/_shared.validators";
export * from "./checkout/checkout.validators";
export * from "./discount/discount.validators";
export * from "./file/file.validators";
export * from "./licenseKey/licenseKey.validators";
export * from "./licenseKeyInstance/licenseKeyInstance.validators";
export * from "./order/order.validators";
export * from "./orderItem/orderItem.validators";
export * from "./product/product.validators";
export * from "./store/store.validators";
export * from "./subscription/subscription.validators";
export * from "./subscriptionInvoice/subscriptionInvoice.validators";
export * from "./user/user.validators";
export * from "./variant/variant.validators";
