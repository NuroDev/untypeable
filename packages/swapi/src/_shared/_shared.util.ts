import { beforeAll } from "vitest";
import { createTypeLevelClient } from "untypeable";
import { fetch } from "undici";
import { join } from "node:path";

import type { SwapiRouter } from "..";

export function useTestClient() {
  const client = createTypeLevelClient<SwapiRouter>(
    async (path, input = {}) => {
      const pathWithParams = path.replace(
        /:([a-zA-Z0-9_]+)/g,
        (_, key) => input[key]
      );

      const url = new URL(join("/api", pathWithParams), "https://swapi.dev/");

      const response = await fetch(url.href, {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
        },
      });

      return await response.json();
    }
  );

  beforeAll(() => {
    if (!client) throw "Failed to initialise untypeable client instance.";
  });

  return client;
}
