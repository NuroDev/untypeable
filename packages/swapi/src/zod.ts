export * from "./root.validators";
export * from "./people.validators";
