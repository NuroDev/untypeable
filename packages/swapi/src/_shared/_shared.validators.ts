import { z } from "zod";

export function defineSWAPIPaginatedSchema<TSchema extends z.ZodTypeAny>(
  dataSchema: TSchema
) {
  return z.object({
    count: z.number(),
    next: z.string().url().nullable(),
    previous: z.string().url().nullable(),
    results: z.array(dataSchema),
  });
}
