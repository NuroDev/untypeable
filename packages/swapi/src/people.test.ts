import { describe, it, expect } from "vitest";

import { PaginatedPersonSchema, PersonSchema } from "./zod";
import { useTestClient } from "./_shared/_shared.util";

describe.concurrent("SWAPI - People", () => {
  const client = useTestClient();

  it("GET - /people", async () => {
    const people = await client("/people");

    expect(people).toBeDefined();
    expect(people.count).toBeTypeOf("number");
    expect(people.next).toBeTypeOf("string");
    expect(people.next).not.toBe(null);
    expect(people.previous).toBe(null);
    expect(Array.isArray(people.results)).toBe(true);
    expect(people.results.length).toBeGreaterThan(0);
    expect(people.results.at(0)).toBeDefined();

    expect(PaginatedPersonSchema.safeParse(people).success).toBe(true);
  });

  it("GET - /people/:id", async () => {
    const person = await client("/people/:id", {
      id: 1,
    });

    expect(person).toBeDefined();
    expect(person).toMatchSnapshot({
      name: "Luke Skywalker",
      height: "172",
      mass: "77",
      hair_color: "blond",
      skin_color: "fair",
      eye_color: "blue",
      birth_year: "19BBY",
      gender: "male",
      homeworld: "https://swapi.dev/api/planets/1/",
      films: [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/6/",
      ],
      species: [],
      vehicles: [
        "https://swapi.dev/api/vehicles/14/",
        "https://swapi.dev/api/vehicles/30/",
      ],
      starships: [
        "https://swapi.dev/api/starships/12/",
        "https://swapi.dev/api/starships/22/",
      ],
      created: "2014-12-09T13:50:51.644000Z",
      edited: "2014-12-20T21:17:56.891000Z",
      url: "https://swapi.dev/api/people/1/",
    });

    expect(PersonSchema.safeParse(person).success).toBe(true);
  });
});
