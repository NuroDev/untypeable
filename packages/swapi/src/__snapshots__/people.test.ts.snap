// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SWAPI - People > GET - /people/:id 1`] = `
{
  "birth_year": "19BBY",
  "created": "2014-12-09T13:50:51.644000Z",
  "edited": "2014-12-20T21:17:56.891000Z",
  "eye_color": "blue",
  "films": [
    "https://swapi.dev/api/films/1/",
    "https://swapi.dev/api/films/2/",
    "https://swapi.dev/api/films/3/",
    "https://swapi.dev/api/films/6/",
  ],
  "gender": "male",
  "hair_color": "blond",
  "height": "172",
  "homeworld": "https://swapi.dev/api/planets/1/",
  "mass": "77",
  "name": "Luke Skywalker",
  "skin_color": "fair",
  "species": [],
  "starships": [
    "https://swapi.dev/api/starships/12/",
    "https://swapi.dev/api/starships/22/",
  ],
  "url": "https://swapi.dev/api/people/1/",
  "vehicles": [
    "https://swapi.dev/api/vehicles/14/",
    "https://swapi.dev/api/vehicles/30/",
  ],
}
`;
